### DISCUSSION TITLE
Regex search tips

### DISCUSSION THREAD

**Discussion by otseng** [eliranwong/UniqueBible#351] [General]
The regex search allows using regex expressions to search for text.  The regex search can be executed from the command line or the Master control.

## Regex cheatsheets

* [Debuggex](https://www.debuggex.com/cheatsheet/regex/python)

* [Devhints](https://devhints.io/regexp)

**Comment by otseng** [comment]
## Example

Find verses that have "son" followed by "David".  

`REGEXSEARCH:::KJVx:::son.*David`

The "." means any character and "*" means any number of these characters.

If you only want to limit it for son using Strongs H1121:

`REGEXSEARCH:::KJVx:::son.*H1121.*David`

However, this will also return:

"son H1121 , and to all his servants H5650 , that they should kill H4191 David H1732."

Use this to limit only up to 10 characters between H1121 and David:

`REGEXSEARCH:::KJVx:::son.*H1121.{0,10}David`

Now search for son using Strongs G5207:

`REGEXSEARCH:::KJVx:::son.*G5207.{0,10}David`

**Reply by darrelwright** [reply]
Hey Thanks Oliver, I appreciate it!
